src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -47,12 +51,8 @@
  */
 public class VCT {
 
-//  public final static class VCTSolver {
```

Comment:
Whether you call it a solver or not, you need a class that gets instantiated and initialized once, that you can reuse over and over, so you don't end up recreating zillions of objects/arrays.